src/main/scala/mesosphere/marathon/core/storage/simple/PersistenceStore.scala

```diff
@@ -0,0 +1,125 @@
+package mesosphere.marathon
+package core.storage.simple
+
+import akka.stream.scaladsl.{Flow, Source}
+import akka.util.ByteString
+import akka.{Done, NotUsed}
+import mesosphere.marathon.core.storage.simple.PersistenceStore.{Node, StoreOp}
+
+import scala.concurrent.Future
+import scala.util.Try
+
+/**
+  * An interface for a simple persistence store abstraction for the underlying Zookeeper store. Supported are basic
+  * create, read, update, delete and check methods along with transaction operations and Zookeeper specific children
+  * and sync methods.
+  */
+trait PersistenceStore {
+
+  /**
+    * A Flow for saving nodes to the store. It takes a stream of nodes and returns a stream of node keys
+    * that were successfully stored.
+    *
+    * By default persistent nodes with empty ACLs are created. If the path is nested parent nodes will be also
```

Comment:
should this really be in the interface description? Since you can actually create the store with any curator with ANY settings?